src/gui/mzroll/spectrawidget.cpp

```diff
@@ -214,36 +211,52 @@ void SpectraWidget::overlayPeptideFragmentation(QString peptideSeq, MassCutoff *
     overlaySpectralHit(hit);
 }
 
-
-/*
-@author: Sahil
-*/
-//TODO: Sahil, Added while merging spectrawidget
-void SpectraWidget::overlayCompoundFragmentation(Compound* c) {
-    if(_currentScan and c->fragmentMzValues.size()) {
-		SpectralHit hit;
-	  	hit.score = 0;
-        hit.precursorMz = c->precursorMz;
-        hit.matchCount=0;
-        hit.sampleName="";
-        hit.productPPM = mainwindow->mavenParameters->fragmentTolerance;
-        hit.scan=NULL;
-        for(int i=0; i < c->fragmentMzValues.size();i++)        hit.mzList << c->fragmentMzValues[i];
-        for(int i=0; i < c->fragmentIntensities.size();i++)  hit.intensityList<< c->fragmentIntensities[i];
-        _spectralHit = hit;//copy hit
-       	cerr << "SpectraWidge::overlayCompoundfragmentation(Compound)" << c->name << " " << c->precursorMz << endl;
-
-       if(_currentScan) {
-            if (_currentScan->mslevel != 1) {
-                 overlaySpectralHit(_spectralHit);
-            }
-       }
-	}
+void SpectraWidget::overlayPeakGroup(PeakGroup* group)
+{
+    if (!group) return;
+    float productPpmTolr = mainwindow->mavenParameters->fragmentTolerance;
+    Scan* avgScan = group->getAverageFragmentationScan(productPpmTolr);
+    setScan(avgScan);
+    if (group->compound)  {
+        overlayCompoundFragmentation(group->compound);
+        //if (!group->compound->smileString.empty()) overlayTheoreticalSpectra(group->compound);
```

Comment:
Since we are delivering an MVP version of the PRM feature, there are certain features from MAVEN that have not been implemented in El-MAVEN yet. The decision to include (or not) this features will be made once the clients have tested it out.